pdns/calidns.cc

```diff
@@ -121,7 +121,7 @@ static ComboAddress getRandomAddressFromRange(const Netmask& ecsRange)
 {
   ComboAddress result = ecsRange.getMaskedNetwork();
   uint8_t bits = ecsRange.getBits();
-  uint32_t mod = 1 << (32 - bits);
+  uint32_t mod = (bits > 0) ? (1 << (32 - bits)) : std::numeric_limits<uint32_t>::max();
   result.sin4.sin_addr.s_addr = result.sin4.sin_addr.s_addr + ntohl(dns_random(mod));
```

Comment:
Pedantic mode: if `mod` is  `std::numeric_limits<uint32_t>::max()` dns_random_uint32() should be used, otherwise 0xffffffff will never be chosen.

Reply:
also since there are dns helpers that are actually verified to return somewhat uniform result, it's no good to reinvent here.

Reply:
> Pedantic mode: if `mod` is `std::numeric_limits<uint32_t>::max()` dns_random_uint32() should be used, otherwise 0xffffffff will never be chosen.

Done!